routes/tp.js

```diff
@@ -339,4 +339,71 @@ router.put('/tp/:id/reserve/now', async function(req, res) {
     res.end();
 });
 
+router.put('/tp/:id/reserve/extend', async function(req, res) {
+    if(! req.locals.logInfo.is_logged) {
+        res.status(403).send({message: 'Not authorized'});
+        return;
+    }
+    if(! sansrv.sanitizeAll([req.params.id])) {
+        res.status(403).send({message: 'Invalid parameters'});
+        return;
+    }
+    let user = null;
+    let isadmin = false;
+    try {
+        user = await dbsrv.mongo_users().findOne({'_id': req.locals.logInfo.id});
+        isadmin = await rolsrv.is_admin(user);
+    } catch(e) {
+        logger.error(e);
+        res.status(404).send({message: 'User session not found'});
+        res.end();
+        return;
+    }
+
+    if(!user) {
+        res.status(404).send({message: 'User does not exist'});  // Why do none of the other routes give a 404 status here ?
+        res.end();
+        return;
+    }
+
+    if(! (isadmin || (user.is_trainer !== undefined && user.is_trainer))) {
+        res.status(403).send({message: 'Not authorized'});
+        return;
+    }
+
+    let reservation_id = ObjectID.createFromHexString(req.params.id);
+
+    let filter = {};
+    if(isadmin) {
+        filter = {_id: reservation_id};
+    }
+    else{
+        filter = {_id: reservation_id, owner: user.uid};
+    }
+    let reservation = await dbsrv.mongo_reservations().findOne(filter);
+    if(!reservation){
+        res.status(403).send({message: 'Not allowed to extend this reservation'});
+        res.end();
+        return;
+    }
+
+    if (req.body.to < reservation.to) {
+        res.status(403).send({message: 'Extended end date must be after current end date'});
+        // Should I add "res.end();" here ?
```

Comment:
Sure

Reply:
Is there a general rule for when we add `res.end();` or not ?

Reply:
Hmm, seems like res.send() calls res.end(), so no need for both. (But there are several place where we call both in the code, maybe it's an error or an old version behaviour?)